src/abstractarray.jl

```diff
@@ -70,24 +69,25 @@ end
 
 # Special case: simple numerical arrays with BLAS-compatible floating point type
 function add!(y::BLASVector{T}, x::BLASVector{T},
-              α::Number=One(), β::Number=One()) where {T<:BlasFloat}
+              α::Number, β::Number) where {T<:BlasFloat}
+    α′ = α === One() ? true : convert(T, α)
```

Comment:
I don't think that line is necessary? The true BLAS methods, to which `LinearAlgebra.axpy!` should lower, require all scalar types to be the same, so `true` is anyway also converted to type `T`.

Reply:
Ah, I didn't know this is how BLAS behaves, I'll fix this

Reply:
I didn't know of this behaviour for BLAS. I guess for the fallbacks we still want to keep the conversion, as this somehow assumes that VectorInterface has not been implemented, and these methods thus might not know about `One`?

Reply:
I updated this here, but left it in the fallbacks, as presumably these methods do not know about the existence of `One` and `Zero`

Reply:
Indeed, there it is different, and we just preserve alpha unless it is of type One.